### PR TITLE
Respect KUBE_BUILD_PLATFORMS set by user

### LINKED ISSUE #20365: KUBE_BUILD_PLATFORMS is not respected
hack/build-cross sets it without looking at the current value.  And the value has to be an array, which is a bit awkward to use.

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

Currently the only(?) toggle available for custom arch builds is
KUBE_FASTBUILD. We should allow the user to specify a list
of arch(es) in the environment variable KUBE_BUILD_PLATFORMS.

Example:
KUBE_BUILD_PLATFORMS="linux/amd64 linux/arm64" hack/build-cross.sh

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes #20365

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cmluciano** [comment]
😊

Related #20367 #20365

**Comment by dims** [comment]
@thockin @lavalamp can you please bless this PR?

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -53,7 +53,12 @@ readonly KUBE_NODE_TARGETS=($(kube::golang::node_targets))
 readonly KUBE_NODE_BINARIES=("${KUBE_NODE_TARGETS[@]##*/}")
 readonly KUBE_NODE_BINARIES_WIN=("${KUBE_NODE_BINARIES[@]/%/.exe}")
 
-if [[ "${KUBE_FASTBUILD:-}" == "true" ]]; then
+if [[ ! -z "${KUBE_BUILD_PLATFORMS:-}" ]]; then
```
Comment (zmerlynn) [inline hack/lib/golang.sh]: nit: This is the same as:
```
if [[ -n "${KUBE_BUILD_PLATFORMS:-}" ]]; then
```
LGTM after.

Comment (dims) [inline hack/lib/golang.sh]: Done thanks! @zmerlynn

**Pull request merged**